### PR TITLE
✨ Implement a dummy Notifier

### PR DESCRIPTION
**Pull request by olibre**

### CODE REVIEW

**Review by kahlys — COMMENTED** [review]
I don't think a package should provider a dummy version of itself. A pkg that a dummy should have one for itself.

**Comment by olibre** [comment]
Thanks for the review @kahlys 
Is it :ok: if we rename `dummy` -> `logger`
The idea is to have a fallback Notifier that just logs the messages.

**Comment by kahlys** [comment]
> Thanks for the review @kahlys Is it 🆗 if we rename `dummy` -> `logger` The idea is to have a fallback Notifier that just logs the messages.

Then it's just a "log" package, not a notifier. I don't think any of this should be in this package

**Comment by 0uep** [comment]
Hey @kahlys

Imagine several apps use the lib `Notifier`. In some circumstances, these apps may not want to send notifications to a real Telegram/Mattermost instance. For example, this can happen when the Telegram/Mattermost URL is missing (in Prod), or when the app is running in a pre-prod environment, or when the developers is just testing the app. Do you agree with that statement?

Thus, these apps require an implementation of the `Notifier` interface that does not fail when the Telegram/Mattermost URL is invalid. Let's say this is the *Dummy* `Notifier` or the *Fallback* `Notifier` if you prefer. Do you still agree with that need?

**Where do you think this *Dummy* `Notifier` should be implemented?**
1. In the repository of all apps
2. In the `Notifier` repository: https://github.com/teal-finance/notifier
3. In a separate repository
4. another idea?

My feeling for these first three alternatives:
1. Developers will redo a similar work each time.
2. The logical repo for the reference implementation of the *Dummy* `Notifier`.
3. To much complexity with another repo containing only the *Dummy* `Notifier`.

**Comment by kahlys** [comment]
> Hey @kahlys
> 
> Imagine several apps use the lib `Notifier`. In some circumstances, these apps may not want to send notifications to a real Telegram/Mattermost instance. For example, this can happen when the Telegram/Mattermost URL is missing (in Prod), or when the app is running in a pre-prod environment, or when the developers is just testing the app. Do you agree with that statement?
> 
> Thus, these apps require an implementation of the `Notifier` interface that does not fail when the Telegram/Mattermost URL is invalid. Let's say this is the _Dummy_ `Notifier` or the _Fallback_ `Notifier` if you prefer. Do you still agree with that need?
> 
> **Where do you think this _Dummy_ `Notifier` should be implemented?**
> 
> 1. In the repository of all apps
> 2. In the `Notifier` repository: https://github.com/teal-finance/notifier
> 3. In a separate repository
> 4. another idea?
> 
> My feeling for these first three alternatives:
> 
> 1. Developers will redo a similar work each time.
> 2. The logical repo for the reference implementation of the _Dummy_ `Notifier`.
> 3. To much complexity with another repo containing only the _Dummy_ `Notifier`.

There is no "logic" in a dummy notifier. It's just one line with a log statement. So such a thing should be writen on the app which need a "dummy" notifier. But, such an app can also handle a 'nil' notifier to avoid using a "dumy notifier".

What are you trying to achieve ? What is the goal ?
If an apps don't want to run with an alerter, just use a 'nil' value.

**Comment by 0uep** [comment]
> What are you trying to achieve ? What is the goal ?

In some circumstances, we do not want the app sends notifications to a real Telegram/Mattermost instance, but we still want to retrieve that notifications from the logs.

Some examples:

1. the app runs in Prod but the Telegram/Mattermost URL is invalid (or missing).
2. the app runs in a CI environment and the official Telegram/Mattermost instance should not receive test-related notifications.
3. the developer is testing the app.
4. ...

See my previous message for more explanations: https://github.com/teal-finance/notifier/pull/3#issuecomment-1162792364

**Comment by kahlys** [comment]
> > What are you trying to achieve ? What is the goal ?
> 
> In some circumstances, we do not want the app sends notifications to a real Telegram/Mattermost instance, but we still want to retrieve that notifications from the logs.
> 
> Some examples:
> 
> 1. the app runs in Prod but the Telegram/Mattermost URL is invalid (or missing).
> 2. the app runs in a CI environment and the official Telegram/Mattermost instance should not receive test-related notifications.
> 3. the developer is testing the app.
> 4. ...
> 
> See my previous message for more explanations: [#3 (comment)](https://github.com/teal-finance/notifier/pull/3#issuecomment-1162792364)

Then it's not a "dummy" notifier but a "logger notifier".

**Comment by 0uep** [comment]
Hello @kahlys
I agree, we may rename it a `logger` notifier.
The renaming has already been proposed in this message https://github.com/teal-finance/notifier/pull/3#issuecomment-1160092263 (last Monday)

> Thanks for the review @kahlys 
> It is :ok: if we rename `dummy` -> `logger`
> The idea is to have a fallback Notifier that just logs the messages.

Please say what you want in order to integrate this type of Notifier within the Git repo.

File: dummy/dummy.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"log"
+)
+
+// Notifier is an empty struct.
+type Notifier struct{}
+
+// NewNotifier creates a new dummy Notifier.
+func NewNotifier(endpoint string) Notifier {
```
Comment (kahlys) [inline dummy/dummy.go]: endpoint unused

**Comment by kahlys** [comment]
> Hello @kahlys I agree, we may rename it a `logger` notifier. The renaming has already been proposed in this message [#3 (comment)](https://github.com/teal-finance/notifier/pull/3#issuecomment-1160092263) (last Monday)
> 
> > Thanks for the review @kahlys
> > It is 🆗 if we rename `dummy` -> `logger`
> > The idea is to have a fallback Notifier that just logs the messages.
> 
> Please say what you want in order to integrate this type of Notifier within the Git repo.

I am ok with a "logger" renaming

File: dummy/dummy.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Package dummy implements a Notifier interface that logs the messages.
+package dummy
```
Comment (kahlys) [inline dummy/dummy.go]: logger

File: logger/logger.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"log"
+)
+
+// Notifier is an empty struct.
+type Notifier struct{}
+
+// NewNotifier creates a new logger Notifier.
+func NewNotifier(_ string) Notifier {
```
Comment (kahlys) [inline logger/logger.go]: func NewNotifier() Notifier {

**Review by kahlys — CHANGES_REQUESTED** [review]

**Review by kahlys — APPROVED** [review]

**Pull request merged**